arxiv/vault/tests/test_manager.py

```diff
@@ -0,0 +1,229 @@
+"""Tests for :mod:`.manager`."""
+
+from unittest import TestCase, mock
+from datetime import datetime
+import time
+from pytz import UTC
+
+from .. import manager
+from ..core import Secret
+
+
+class TestGetSecretsNotAuthenticated(TestCase):
+    """We use a :class:`.SecretsManager` to grab Vault secrets."""
+
+    def setUp(self):
+        """We have a :class:`.Vault` connection and are not authenticated."""
+        self.vault = mock.MagicMock(authenticated=False)
+
+    def test_generic_request(self):
+        """The app requires a generic secret."""
+        requests = [
+            manager.SecretRequest.factory('generic', **{
```

Comment:
Just wondering: does this request mean "get me type: generic key: GENERIC_FOO"?

What do the path, key and mount_point do?